packages/open-next/src/adapters/routing/middleware.ts

```diff
@@ -137,8 +137,13 @@ export async function handleMiddleware(
   // NOTE: the header was added to `req` from above
   const rewriteUrl = responseHeaders.get("x-middleware-rewrite");
   let rewritten = false;
+  let middlewareQueryString = internalEvent.query;
   if (rewriteUrl) {
-    req.url = new URL(rewriteUrl).pathname;
+    const rewriteUrlObject = new URL(rewriteUrl);
+    req.url = rewriteUrlObject.pathname;
+    rewriteUrlObject.searchParams.forEach((v: string, k: string) => {
+      middlewareQueryString[k] = v;
```

Comment:
Thx! I'm wondering if we need to support the case where the url's search params are a list of items, eg:
`?name=Bob&name=Alice`... some applications treat that as `name = [Bob, Alice]`.
I think we should support this scenario.

Reply:
Can you create another ticket for supporting search param list if you want to defer that feature to another PR.
That way we can get this in for 2.2.4

Reply:
Hi @khuezy,

Sure, I can create another support ticket for handling URL search parameters. 
Let me know if I can proceed or we want to wait the discussion with @conico974.
Regarding the desired output, I've experimented with how "plain" next handles URLs in this form, like "?name=Bob&name=Alice," and it appears that it simply combines the values by appending them. Specifically, within the page, I receive a single string value, 'BobAlice' .

Reply:
If next doesn't support lists then we shouldn't either.